lib/matplotlib/axis.py

```diff
@@ -1012,6 +1012,7 @@ def _set_artist_props(self, a):
             return
         a.set_figure(self.figure)
 
+    @cbook.deprecated("3.1")
     def iter_ticks(self):
```

Comment:
This looks great, but I wonder if you want a private `_locate_ticks` and then call it here, and in the `_update_ticks`.  That way the (deprecated) public function doesn't drift from the private implimentation?

Reply:
But then (hypothetical, but that's what this discussion is about) changes in _locate_ticks would be reflected in iter_ticks, and you'd need to add additional changelog entries for changes in the behavior of the deprecated method, which seems a bit silly.